tests/unit/test_patch_models.py

```diff
@@ -105,3 +105,35 @@ async def test_patch_database_model_with_nested_models(self) -> None:
             org_get = await Org.get_with_refs(org.id)
             assert org_get.name == "new test name"
             assert org_get.tech_owner.username == "bberkley"
+
+    @pytest.mark.asyncio
+    async def test_patch_database_model_with_delete(self) -> None:
+        """Test that we can use `delete` on a patched db model"""
+        orgs = [
+            Org(id=str(uuid4()), name="orgus borgus", slug="orgus_borgus"),
+            Org(id=str(uuid4()), name="orgus borgus2", slug="orgus_borgus"),
+            Org(id=str(uuid4()), name="nonorgus borgus", slug="orgus_borgus"),
+        ]
+
+        with patch_database_model(Org, models=orgs):
+            orgs = await Org.get_list()
+            assert len(orgs) == 3
+            await orgs[0].delete()
+            orgs = await Org.get_list()
+            assert len(orgs) == 2
```

Comment:
For these two tests could you inspect the remaining list to ensure the deletion targeted the right keys?

Reply:
updated